src/main/kotlin/net/leanix/githubagent/services/GitHubWebhookHandler.kt

```diff
@@ -0,0 +1,50 @@
+package net.leanix.githubagent.services
+
+import net.leanix.githubagent.config.GitHubEnterpriseProperties
+import net.leanix.githubagent.exceptions.InvalidEventSignatureException
+import net.leanix.githubagent.exceptions.WebhookSecretNotSetException
+import net.leanix.githubagent.shared.SUPPORTED_EVENT_TYPES
+import net.leanix.githubagent.shared.hmacSHA256
+import net.leanix.githubagent.shared.timingSafeEqual
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+
+@Service
+class GitHubWebhookHandler(
+    private val webhookEventService: WebhookEventService,
+    private val gitHubEnterpriseProperties: GitHubEnterpriseProperties
+) {
+
+    private val logger = LoggerFactory.getLogger(GitHubWebhookHandler::class.java)
+    private var isWebhookProcessingEnabled = true
+
+    fun handleWebhookEvent(eventType: String, host: String, signature256: String?, payload: String) {
+        if (!isWebhookProcessingEnabled) {
+            throw WebhookSecretNotSetException()
+        }
```

Comment:
Question: Why don't we remove this block and also remove `isWebhookProcessingEnabled`?